### ISSUE TITLE
Inquiry about nn.Module's `get_buffer`, `register_buffer`, and `register_parameter` equivalents in mlx.

### ISSUE THREAD

**Issue by m0saan** [ml-explore/mlx#141]
I am looking for functionalities equivalent to PyTorch's `nn.Module` methods: `get_buffer(target)`, `register_buffer(name, tensor, persistent=True)`, and `register_parameter(name, param)`.

### Requested Information

1. **`get_buffer(target)` Equivalent:**
   - Does MLX provide a method similar to PyTorch's `get_buffer(target)`? This method is used to retrieve a buffer by its name or type.

2. **`register_buffer(name, tensor, persistent=True)` Equivalent:**
   - This method is used to register a persistent buffer, typically a non-parameter tensor.

3. **`register_parameter(name, param)` Equivalent:**
   - This method is used to register a learnable parameter within a module.

**Comment by angeloskath** [comment]
If I may point you to #127 for the full reply.

In short, all member `arrays` are parameters. Arrays can be "frozen" to become buffers. Private members are treated as constants meaning that they are not returned neither by `trainable_parameters()` nor by `parameters()`.

I will close the issue but feel free to reopen it if you think that my reply above doesn't cover your question.

**Comment by m0saan** [comment]
@angeloskath Thank you for your prompt response, and I appreciate the information provided in #127. Regarding model parameters that need to be saved and restored in the state dict, an illustrative example of a buffer can be found in the [_BatchNorm](https://pytorch.org/docs/1.1.0/_modules/torch/nn/modules/batchnorm.html#BatchNorm2d) module. In this module, the `running_mean`, `running_var`, are registered as buffers and get updated by accumulating statistics from the data forwarded through the layer. this information in the `running_mean`, `running_var` are then used when doing inference or when the model in `eval` mode.  `register_buffer(name, tensor)` can be used in that case, WDYT?
